app/controllers/comments_controller.rb

```diff
@@ -1,4 +1,19 @@
 class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build(comment_params)
+    @comment.user_id = current_user.id
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to user_post_path(current_user, @post) }
+        flash[:success] = 'Comment created successfully'
+      else
+        flash[:error] = 'Failed to save comment'
+        format.html { redirect_to user_post_path(current_user, @post) }
+      end
+    end
+  end
```

Comment:
- [ ] in the commentsController there should be a `new` action as well.

Reply:
Oh thanks @mikethreels, I forgot to add that. Doing that right away